sound/soc/sof/topology.c

```diff
@@ -2031,6 +2059,15 @@ static int sof_dai_load(struct snd_soc_component *scomp, int index,
 		goto free_playback_tables;
 	}
 
+	/* bind pcm to host comp */
+	ret = spcm_bind(sdev, spcm, stream);
+	if (ret) {
+		dev_err(sdev->dev,
+			"error: can't bind pcm to host\n");
+		snd_dma_free_pages(&spcm->stream[stream].page_table);
+		goto free_playback_tables;
```

Comment:
Should we also free capture tables here?

Reply:
@bardliao 		snd_dma_free_pages(&spcm->stream[stream].page_table); frees the capture tables before freeing the playback tables

Reply:
@ranj063 Oh, yes. Didn't notice there is a snd_dma_free_pages before the goto statement :)